autowarefoundation/autoware.universe
# feat(diagnostic_graph_aggregator): add dump tool

## Description

Add dump tool for diagnostic_graph_aggregator.

## Tests performed

1. Run this command
    `ros2 launch diagnostic_graph_aggregator example.launch.xml`
2. Run this command in another terminal
    `ros2 run diagnostic_graph_aggregator dump`
3. Check that you see output like below
    ```
    | ----- | ----- | -------------------------------- | ----- |
    | index | level | name                             | links |
    | ----- | ----- | -------------------------------- | ----- |
    | 0     | OK    | /sensing/radars/front            |       |
    | 1     | OK    | /sensing/lidars/front            |       |
    | 2     | ERROR | /sensing/lidars/top              |       |
    | 3     | OK    | /functions/obstacle_detection    | 1 0   |
    | 4     | ERROR | /functions/pose_estimation       | 2     |
    | 5     | OK    | /external/remote_command         |       |
    | 6     | OK    | /external/joystick_command       |       |
    | 7     | ERROR | /autoware/modes/pull_over        | 4 3   |
    | 8     | OK    | /autoware/modes/comfortable_stop | 3     |
    | 9     | OK    | /autoware/modes/emergency_stop   |       |
    | 10    | OK    | /autoware/modes/remote           | 5     |
    | 11    | OK    | /autoware/modes/local            | 6     |
    | 12    | ERROR | /autoware/modes/autonomous       | 4 3   |
    | 13    | OK    | /autoware/modes/stop             |       |
    ```

## Effects on system behavior



Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.


## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.


## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.


After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/
